pkg/ingester/errors.go

```diff
@@ -51,14 +52,29 @@ type errorWithStatus struct {
 	status *status.Status
 }
 
-func newErrorWithStatus(err error, code int) errorWithStatus {
+func newErrorWithStatus(err error, code codes.Code) errorWithStatus {
 	return errorWithStatus{
 		err:    err,
-		status: status.New(codes.Code(code), err.Error()),
+		status: status.New(code, err.Error()),
+	}
+}
+
+func newErrorWithHTTPStatus(err error, code int) errorWithStatus {
+	errWithHTTPStatus := httpgrpc.Errorf(code, err.Error())
+	stat, ok := status.FromError(errWithHTTPStatus)
+	if !ok {
+		stat = nil
+	}
```

Comment:
Can this happen? I think it can't.

Reply:
If you don't trust it to be correctly implemented, you can just panic instead of proceeding while ignoring the `ok`.

Reply:
Anyway, this error type should live in `httpgrpc` to allow us ignore the implementation details.